Kernel/API/POSIX/sys/socket.h

```diff
@@ -81,6 +82,32 @@ struct msghdr {
     int msg_flags;
 };
 
+// These three are non-POSIX, but common:
+#define CMSG_ALIGN(x) (((x) + sizeof(void*) - 1) & ~(sizeof(void*) - 1))
+#define CMSG_SPACE(x) (CMSG_ALIGN(sizeof(struct cmsghdr)) + CMSG_ALIGN(x))
+#define CMSG_LEN(x) (CMSG_ALIGN(sizeof(struct cmsghdr)) + (x))
+
+static inline struct cmsghdr* CMSG_FIRSTHDR(struct msghdr* msg)
+{
+    if (msg->msg_controllen < sizeof(struct cmsghdr))
+        return 0;
```

Comment:
shouldn't this also return `NULL` (or whatever the solution for the above comment ends up being) to be consistent with `CMSG_NXTHDR`?

Reply:
ah, it's moved from somewhere else. Might still be nice to update it :)